app/Models/Multiplayer/Room.php

```diff
@@ -33,12 +34,66 @@ class Room extends Model
 {
     use SoftDeletes;
 
+    const SORTS = [
+        'ended' => [
+            ['column' => 'ends_at', 'order' => 'desc', 'type' => 'time'],
+            ['column' => 'id', 'order' => 'desc', 'type' => 'int'],
+        ],
+        'created' => [
+            ['column' => 'id', 'order' => 'desc', 'type' => 'int'],
+        ],
+    ];
+
     protected $table = 'multiplayer_rooms';
     protected $dates = ['starts_at', 'ends_at'];
     protected $attributes = [
         'participant_count' => 0,
     ];
 
+    public static function search($params, $preloads = null, $includes = null)
+    {
+        $query = static::query();
+
+        $mode = presence(get_string($params['mode'] ?? null));
+        $user = $params['user'];
+        $sort = 'created';
+
+        switch ($mode) {
+            case 'ended':
+                $query->ended();
+                $sort = 'ended';
+                break;
+            case 'participated':
+                $query->hasParticipated($user);
+                break;
+            case 'owned':
+                $query->startedBy($user);
+                break;
+            default:
+                $query->active();
+        }
+
+        $category = presence(get_string($params['category'] ?? null));
```

Comment:
Apparently the plan is not to have multiple categories in the same list, so this should default to `normal` or something (also solves issue of index not being used when ordering by `ended_at`)